### ISSUE TITLE
RRULE parsing fails when BYDAY specifies a numeric modifier

### ISSUE THREAD

**Issue by aggregat4** [fmeringdal/rust_rrule#4]
Using rrule 0.3.4 it fails parsing RRULE statements that contain BYDAY specifiers with a numeric modifier. For example:

``` 
Error building ruleset from RRULE:FREQ=MONTHLY;UNTIL=20210504T154500Z;INTERVAL=1;BYDAY=1TU
DTSTART:20200901T174500: Encountered parsing error: Invalid weekday: 1TU
Error building ruleset from RRULE:FREQ=MONTHLY;UNTIL=20210504T220000Z;INTERVAL=1;BYDAY=1WE
DTSTART;VALUE=DATE:20200902: Encountered parsing error: Invalid weekday: 1WE
Error building ruleset from RRULE:FREQ=MONTHLY;UNTIL=20210505T080000Z;INTERVAL=1;BYDAY=1WE
DTSTART:20200902T100000: Encountered parsing error: Invalid weekday: 1WE
Error building ruleset from RRULE:FREQ=MONTHLY;UNTIL=20210505T143000Z;INTERVAL=1;BYDAY=1WE
DTSTART:20200902T163000: Encountered parsing error: Invalid weekday: 1WE
Error building ruleset from RRULE:FREQ=MONTHLY;UNTIL=20210524T090000Z;INTERVAL=1;BYDAY=4MO
DTSTART:20201026T110000: Encountered parsing error: Invalid weekday: 4MO
```

**Comment by fmeringdal** [comment]
**UPDATE**

I have implemented the solution now and done some initial testing. This will be released in 0.3.5 after more testing is done, which I will perform in a couple of days :+1:

**Comment by fmeringdal** [comment]
This should now have been fixed in 0.3.5. Thanks for reporting the issue!
